clang/include/clang/Driver/Driver.h

```diff
@@ -718,6 +718,28 @@ class Driver {
     return IsOffload ? OffloadLTOMode : LTOMode;
   }
 
+  /// FPGA Emulation Mode.  By default, this is true due to the fact that
+  /// an external option setting is required to target hardware.
+  enum DeviceMode {
+    FPGAHWMode,
+    FPGAEmulationMode
+  } OffloadCompileMode = FPGAEmulationMode;
+
+  bool IsFPGAHWMode() const { return OffloadCompileMode == FPGAHWMode; }
+
+  bool IsFPGAEmulationMode() const {
+    return OffloadCompileMode == FPGAEmulationMode;
+  }
+
+  void setOffloadCompileMode(StringRef ModeVale) {
```

Comment:
Use of a string to set the mode feels off, and a bit arbitrary.  Why not just set with the enumerated value?